src/browser/api/window.js

```diff
@@ -1319,16 +1319,40 @@ Window.setWindowPreloadState = function(identity, payload) {
     } // @TODO ofEvents.emit(route.frame for iframes
 };
 
-Window.getSnapshot = function(identity, callback = () => {}) {
-    let browserWindow = getElectronBrowserWindow(identity);
+Window.getSnapshot = (opts) => {
+    return new Promise((resolve, reject) => {
+        const { identity, payload: { area } } = opts;
+        const browserWindow = getElectronBrowserWindow(identity);
+
+        if (!browserWindow) {
+            const error = new Error(`Unknown window named '${identity.name}'`);
+            return reject(error);
+        }
 
-    if (!browserWindow) {
-        callback(new Error(`Unknown window named '${identity.name}'`));
-        return;
-    }
+        const callback = (img) => {
+            const imageBase64 = img.toPNG().toString('base64');
+            resolve(imageBase64);
+        };
+
+        if (typeof area === 'undefined') {
+            // Snapshot of a full window
+            return browserWindow.capturePage(callback);
+        }
+
+        if (
+            !area ||
+            typeof area !== 'object' ||
+            typeof area.x !== 'number' ||
+            typeof area.y !== 'number' ||
+            typeof area.width !== 'number' ||
+            typeof area.height !== 'number'
+        ) {
+            const error = new Error(`Invalid shape of the snapshot's area.`);
+            return reject(error);
+        }
 
-    browserWindow.capturePage(img => {
-        callback(undefined, img.toPNG().toString('base64'));
+        // Snapshot of a specified area of the window
+        browserWindow.capturePage(area, callback);
```

Comment:
what happens if numbers in area are not valid for the window.

Reply:
Only the visible portion of that area will be returned.
E.g.: your window is 100px wide, but you asked for 200px, you will only get 100px

Reply:
I mean if width is 100 and area.x has 10000000000000000000000

Reply:
In that case - empty string

Reply:
If anything bad happens that electron doesn't support and errors out - you'll get it in the error callback